p2panda-rs/src/secret_group/mls/provider.rs

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use memory_keystore::MemoryKeyStore;
+use openmls_traits::OpenMlsCryptoProvider;
+use rust_crypto::RustCrypto;
+
+/// Implements the `OpenMlsCryptoProvider` trait to be used as a crypto and key store backend for
+/// all other MLS structs and methods.
+///
+/// @TODO: This will use our own key store soon.
+#[derive(Debug)]
+pub struct MlsProvider {
```

Comment:
For me the provider is maybe the hardest struct to grasp conceptually. Maybe the docs could be expanded to include some more info about it's uses and also maybe it's role in MLS architecture and how we are implementing it in p2panda?